db/data.py

```diff
@@ -92,6 +95,47 @@ def update_spot(spot_id, spotName, spotAddress, spotCapacity, spotImage):
     return response
 
 
+def update_individual_spot_factor(spot_id, factorName, factorRating):
+    if not (1 <= factorRating <= 10):
+        return
+    spotFactor = dbc.get_spot_factor(spot_id, factorName)
+    today = datetime.today().date()
+
+    spot_document = {}
+    spot_document["spotUpdate"] = str(datetime.datetime.now())
+    
+    # if dates are not the same, then new day has begun, need to reset data with current input
+    if datetime.strptime(spotFactor["factorDate"], '%Y-%m-%d').date() != today:
+        print("Wrong date")
+        spot_document[factorName] = {
+            "factorDate": datetime.today().date().strftime('%Y-%m-%d'),
+            "factorValue": factorRating,
+            "factorNumOfInputs": 1
+        }
+        dbc.update_spot_factor(spot_id, spot_document)
+    # if day is still going on, need to average out new input
+    else:
+        print("Correct date")
+        # print(datetime.today().date())
+        # print(datetime.strptime(spotFactor["factorDate"], '%Y-%m-%d').date())
+        currentRating = spotFactor["factorValue"]
```

Comment:
I added a rating element to reviews. Should I take that out of reviews since `factorValue` is kinda like a rating? Or should `factorValue` be taken out? Seems redundant to have both

Reply:
I think ratings have different meanings, we can keep both? For factor, the rating is like out of 10 how full is this space today and for reviews, the rating is just like how many stars in general would you give for this place.

Reply:
Ohhh that makes sense. Yeah, both should be kept. Another question - how is the `factorRating` for `factorAvailability` different from the `spotCapacity` field on a spot object?

Reply:
Like would `spotCapacity` be determined by us based on the `factorRating` for `factorAvailability`?

Reply:
`spotCapacity` would be like how an typical array has a capacity (max-space), but `factorAvailability` would be the length of array (how many seats are currently taken up).